kernel: bump 6.1 to 6.1.44

graysky2 opened the pull request:
Changelog: https://lore.kernel.org/stable/2023080822-repost-unfiled-2f01@gregkh/

All patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod

graysky2 commented:
@john-tho - thoughts on the mediatek, filogic failure?  Due to https://github.com/openwrt/openwrt/commit/36d0aa9c2db0f74614b04a152fb092bcc74afb8e ?

john-tho commented:
> @john-tho - thoughts on the mediatek, filogic failure?

Error: https://github.com/openwrt/openwrt/actions/runs/5754182182/job/15601799801
tail of log:
>```
>drivers/pinctrl/mediatek/pinctrl-mt7988.c:1402:18: error: initialization discards 'const' qualifier >from pointer target type [-Werror=discarded-qualifiers]
> 1402 |         { "udi", mt7988_udi_groups, ARRAY_SIZE(mt7988_udi_groups) },
>      |                  ^~~~~~~~~~~~~~~~~
>drivers/pinctrl/mediatek/pinctrl-mt7988.c:1403:18: error: initialization discards 'const' qualifier >from pointer target type [-Werror=discarded-qualifiers]
> 1403 |         { "usb", mt7988_usb_groups, ARRAY_SIZE(mt7988_usb_groups) },
>      |                  ^~~~~~~~~~~~~~~~~
>drivers/pinctrl/mediatek/pinctrl-mt7988.c:1404:18: error: initialization discards 'const' qualifier >from pointer target type [-Werror=discarded-qualifiers]
> 1404 |         { "led", mt7988_led_groups, ARRAY_SIZE(mt7988_led_groups) },
>      |                  ^~~~~~~~~~~~~~~~~
>```
File is: https://github.com/openwrt/openwrt/blob/main/target/linux/mediatek/files-6.1/drivers/pinctrl/mediatek/pinctrl-mt7988.c

I have not spotted an upstream change that caused this (https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/?h=linux-6.1.y&id=v6.1.42&id2=v6.1.43). If so, the fix should be a separate commit. This is one for @dangowrt.
Looking at other upstream pinctrl drivers, I only see [`base_names` used as `const char * const`](https://elixir.bootlin.com/linux/v6.1.43/source/drivers/pinctrl/mediatek/pinctrl-mtk-common-v2.h#L248) (const pointer), I did not find any [function groups using this](https://elixir.bootlin.com/linux/v6.1.43/C/ident/function_desc).

---

It is worth mentioning the config change for the errata both here, and in the commit message: [6df696cd9bc1 ("arm64: errata: Mitigate Ampere1 erratum AC03_CPU_38 at stage-2")](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.1.y&id=cde7f2fd6323e5054e1db2f1ee7231864cd5294c). So that someone can check if it is needed for any targets. Looks like these get disabled in generic config.

graysky2 commented:
> It is worth mentioning the config change for the errata both here, and in the commit message: [6df696cd9bc1 ("arm64: errata: Mitigate Ampere1 erratum AC03_CPU_38 at stage-2")](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.1.y&id=cde7f2fd6323e5054e1db2f1ee7231864cd5294c). So that someone can check if it is needed for any targets. Looks like these get disabled in generic config.

I found that it is needed for qualcommax/ipq807x and at least one more, filogic/xiaomi_redmi-router-ax6000-ubootmod (testing kernel enabled).  Is there a systematic method to determine if others are needed?

john-tho commented:
> > It is worth mentioning the config change for the errata both here, and in the commit message: [6df696cd9bc1 ("arm64: errata: Mitigate Ampere1 erratum AC03_CPU_38 at stage-2")](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.1.y&id=cde7f2fd6323e5054e1db2f1ee7231864cd5294c). So that someone can check if it is needed for any targets. Looks like these get disabled in generic config.
> 
> I found that it is needed for qualcommax/ipq807x and at least one more, filogic/xiaomi_redmi-router-ax6000-ubootmod (testing kernel enabled). Is there a systematic method to determine if others are needed?

The prompt showing up as unconfigured does not indicate that this is _needed_, only that is may be applicable (target is arm64). qualcommax does not use Ampere1 CPUs, nor does filogic, so they do not need it.

john-tho commented:
> > The prompt showing up as unconfigured does not indicate that this is needed, only that is may be applicable (target is arm64). qualcommax does not use Ampere1 CPUs, nor does filogic, so they do not need it.
> 
> Their respective builds fail if I do not set it. For example, filogic:
> 
> ```
> % make V=s
> ...
> make[5]: Entering directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.43'
>   SYNC    include/config/auto.conf
> make[6]: Entering directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.43'
> make[7]: Entering directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.43'
> *
> * Restart config...
> *
> *
> * ARM errata workarounds via the alternatives framework
> *
> AmpereOne: AC03_CPU_38: Certain bits in the Virtualization Translation Control Register and Translation Control Registers do not follow RES0 semantics (AMPERE_ERRATUM_AC03_CPU_38) [Y/n/?] (NEW) 
> ```



> Looks like these get disabled in generic config.

Example: https://github.com/openwrt/openwrt/blob/1eb67cb0704a61f49d9be9f88aecbce1d881f8c9/target/linux/generic/config-6.1#L376

robimarko commented:
Please dont enable the Ampere errata fix, `qualcommax` nor any other target in OpenWrt requires it so like @john-tho pointed out it should just be disabled in generic config.

graysky2 commented:
Looks like just qualcommax is failing due to that symbol not getting set by the generic config.  I pushed a commit that sets it to disabled in `target/linux/qualcommax/config-6.1` ... if that builds successfully are we good to keep the commit?  Let me know if I should squash it into the same bump to 6.1.43 if so.

graysky2 commented:
Got it, see recent commit

robimarko commented on target/linux/qualcommax/config-6.1:
You probably forgot to drop this

graysky2 replied:
head smack

Done

graysky2 committed:
kernel: bump 6.1 to 6.1.44

Changelog: https://lore.kernel.org/stable/2023080822-repost-unfiled-2f01@gregkh/

All patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

DragonBluep commented on target/linux/bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:
```diff
@@ -22,17 +22,17 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		xhci->quirks |= XHCI_LPM_SUPPORT;
- 		xhci->quirks |= XHCI_EP_CTX_BROKEN_DCS;
 +		xhci->quirks |= XHCI_AVOID_DQ_ON_LINK;
- 	}
```
Parentheses are still necessary here.

graysky2 replied:
Would you mind making the needed change and pushing a commit to my branch so I can pull it please?

john-tho replied:
Good catch. This is for the 6.1.43 commit.
The `if` needs curly brackets added `{`,`}` because there is more than one line to be run:
```
	if (pdev->vendor == PCI_VENDOR_ID_VIA && pdev->device == 0x3483) {
		xhci->quirks |= XHCI_LPM_SUPPORT;
		xhci->quirks |= XHCI_AVOID_DQ_ON_LINK;
	}
```
There are a few follow-up patches to the quirks for this device, so they will need refresh as well (your manually rebased list)

graysky2 replied:
Thanks it is clear to me now.

graysky2 replied:
Fixed, please verify.

graysky2 committed:
kernel: bump 6.1 to 6.1.43

1. Disable Ampere errata fix in target/linux/generic/config-6.1

2. Update kernel Changelog: https://lore.kernel.org/stable/2023061431-modular-data-8489@gregkh/

Manually rebased:
        bcm27xx/patches-6.1/950-0359-xhci-quirks-add-link-TRB-quirk-for-VL805.patch
        bcm27xx/patches-6.1/950-0362-usb-xhci-add-VLI_TRB_CACHE_BUG-quirk.patch
        bcm27xx/patches-6.1/950-0390-usb-xhci-add-a-quirk-for-Superspeed-bulk-OUT-transfe.patch
        bcm27xx/patches-6.1/950-0469-usb-xhci-add-XHCI_VLI_HUB_TT_QUIRK.patch
        bcm27xx/patches-6.1/950-0520-xhci-constrain-XHCI_VLI_HUB_TT_QUIRK-to-old-firmware.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: x86/64/AMD Cezanne, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>
